dcmtk 3.6.7 (or later) errors on use of Macro INCLUDE_CSTDIO etc

We get Gadgetron [compilation problems on Ubuntu 24.04](https://github.com/SyneRBI/SIRF/actions/runs/12877252649/job/35901418387?pr=1301#step:8:9330)
```
In file included from /usr/include/dcmtk/oflog/config.h:31,
                 from /usr/include/dcmtk/oflog/logger.h:28,
                 from /usr/include/dcmtk/oflog/oflog.h:28,
                 from /usr/include/dcmtk/dcmdata/dctypes.h:27,
                 from /usr/include/dcmtk/dcmdata/dctk.h:28,
                 from /home/runner/work/SIRF/SIRF/build/sources/Gadgetron/gadgets/dicom/DicomFinishGadget.h:24,
                 from /home/runner/work/SIRF/SIRF/build/sources/Gadgetron/gadgets/dicom/DicomFinishGadget.cpp:4:
/usr/include/dcmtk/ofstd/ofstdinc.h:113:2: error: #error "Macro INCLUDE_CSTDIO not supported anymore. Include <cstdio> directly."
```
This is caused by a change in DCMTK 3.6.7 that converts previous macros into errors
https://github.com/DCMTK/dcmtk/blob/a137f1aff4e1df3fbefe53ee8b160973c74c96dd/ofstd/include/dcmtk/ofstd/ofstdinc.h#L112-L114 (commited on DCMTK master on  Apr 4, 2021).

We define these macros here
https://github.com/gadgetron/gadgetron/blob/846ed3fb57b235390ab0fb953edd17af86935d61/gadgets/dicom/DicomFinishGadget.h#L21-L23
(and in 2 other places). 

It appears that this needs something like
```c++
#if DCMTK_VERSION < 3.6.7 // this is not a correct check obviously 
 #define INCLUDE_CSTDLIB 
 #define INCLUDE_CSTDIO 
 #define INCLUDE_CSTRING 
#else
#include <cstdlib>
#include <cstdio>
#include <cstring>
#endif
```
Or can we just delete those 3 lines?

--- a/gadgets/dicom/DicomFinishGadget.h
+++ b/gadgets/dicom/DicomFinishGadget.h
@@ -18,9 +18,9 @@ The dicom image is sent out with message id -> dicom image -> dicom image name -
 
 #include "dcmtk/config/osconfig.h"
 #include "dcmtk/ofstd/ofstdinc.h"
-#define INCLUDE_CSTDLIB
-#define INCLUDE_CSTDIO
-#define INCLUDE_CSTRING
+#include <cstdlib>
+#include <cstdio>
+#include <cstring>
 #include "dcmtk/dcmdata/dctk.h"
 #include "dcmtk/dcmdata/dcostrmb.h"
 

--- a/gadgets/dicom/DicomImageWriter.cpp
+++ b/gadgets/dicom/DicomImageWriter.cpp
@@ -10,9 +10,9 @@
 #include "ismrmrd/meta.h"
 
 // DCMTK includes
-#define INCLUDE_CSTDLIB
-#define INCLUDE_CSTDIO
-#define INCLUDE_CSTRING
+#include <cstdlib>
+#include <cstdio>
+#include <cstring>
 #define NOGDI
 #include "dcmtk/config/osconfig.h"
 #include "dcmtk/ofstd/ofstdinc.h"

--- a/gadgets/dicom/dicom_ismrmrd_utility.h
+++ b/gadgets/dicom/dicom_ismrmrd_utility.h
@@ -14,9 +14,9 @@
 
 #include "dcmtk/config/osconfig.h"
 #include "dcmtk/ofstd/ofstdinc.h"
-#define INCLUDE_CSTDLIB
-#define INCLUDE_CSTDIO
-#define INCLUDE_CSTRING
+#include <cstdlib>
+#include <cstdio>
+#include <cstring>
 #include "dcmtk/dcmdata/dctk.h"
 #include "dcmtk/dcmdata/dcostrmb.h"
 

